src/app/modules/form/public-form/__tests__/public-form.controller.spec.ts

```diff
@@ -389,4 +421,741 @@ describe('public-form.controller', () => {
       expect(mockRes.redirect).toHaveBeenCalledWith(expectedRedirectPath)
     })
   })
+
+  describe('handleGetPublicForm', () => {
+    const MOCK_FORM_ID = new ObjectId().toHexString()
+    const MOCK_USER_ID = new ObjectId().toHexString()
+    const MOCK_USER = {
+      _id: MOCK_USER_ID,
+      email: 'randomrandomtest@example.com',
+    } as IPopulatedUser
+
+    const MOCK_SCRUBBED_FORM = ({
+      _id: MOCK_FORM_ID,
+      title: 'mock title',
+      admin: { _id: MOCK_USER_ID },
+    } as unknown) as PublicForm
+
+    const BASE_FORM = {
+      admin: MOCK_USER,
+      _id: MOCK_FORM_ID,
+      title: MOCK_SCRUBBED_FORM.title,
+      getUniqueMyInfoAttrs: jest.fn().mockReturnValue([MyInfoAttribute.Name]),
+      getPublicView: jest.fn().mockReturnValue(MOCK_SCRUBBED_FORM),
+    }
+
+    const MOCK_REQ = expressHandler.mockRequest({
+      params: {
+        formId: MOCK_FORM_ID,
+      },
+    })
+
+    const MOCK_MYINFO_COOKIE = {
+      accessToken: 'cookie',
+      usedCount: 0,
+      state: MyInfoCookieState.Success,
+    }
+
+    let MOCK_REQ_WITH_COOKIES: ReturnType<typeof expressHandler.mockRequest>
+
+    beforeEach(() => {
+      MOCK_REQ_WITH_COOKIES = expressHandler.mockRequest({
+        params: {
+          formId: MOCK_FORM_ID,
+        },
+        others: { cookies: { MyInfoCookie: MOCK_MYINFO_COOKIE } },
+      })
+    })
+
+    // Success
+    describe('valid form id', () => {
+      const MOCK_JWT_PAYLOAD: JwtPayload = {
+        userName: 'mock',
+        rememberMe: false,
+      }
+
+      beforeAll(() => {
+        MockFormService.setIsIntranetFormAccess.mockImplementation(
+          (_, publicForm) => ok({ ...publicForm, isIntranetUser: false }),
+        )
+
+        MockFormService.setMyInfoError.mockImplementation((publicForm) =>
+          ok({ ...publicForm, myInfoError: false }),
+        )
+      })
+
+      it('should return 200 when there is no AuthType on the request', async () => {
+        // Arrange
+        const MOCK_NIL_AUTH_FORM = (mocked({
```

Comment:
nit: since this is not a "global" constant, use camelCase instead of UPPER_SNAKE?

Reply:
do we really need to use the `mocked` util for this? Can we not just create a new object?  (Same for all other instantiations like this)

```ts
const mockNilAuthForm = {
  ...BASE_FORM,
  authType: AuthType.NIL,
}
```